tgui/packages/tgui/interfaces/ListInputModal.tsx

```diff
@@ -32,56 +33,60 @@ export const ListInputModal = (_, context) => {
     'searchQuery',
     ''
   );
-  const [inputIsValid, setInputIsValid] = useLocalState<Validator>(
-    context,
-    'inputIsValid',
-    { isValid: true, error: null }
-  );
   // User presses up or down on keyboard
   // Simulates clicking an item
   const onArrowKey = (key: number) => {
     const len = filteredItems.length - 1;
     if (key === KEY_DOWN) {
       if (selected === null || selected === len) {
         onClick(0);
+        document!.getElementById('0')?.scrollIntoView();
```

Comment:
This is sort of terrifying, why are we directly indexing things based on ID? Can you not use refs instead? Or at least a better id than just the number?

Reply:
I could index them by the name but it's going to be a string almost regardless. And I am not sure that I can use refs in SFC without a callback function of some type. I don't think this would make it any more performant, as there will be no duplicate indexes or entries as those are handled in DM